### PR TITLE
[SPARK-19043][SQL]Make SparkSQLSessionManager more configurable

### PR DESCRIPTION
**Pull request by yaooqinn**
## What changes were proposed in this pull request?

To make SparkSQLSessionManager's background operation thread pool not only configurable for pool size but also the wait queue size and the thread keep alive time.  To be consistent with HiveServer2's behavior.

## How was this patch tested?
manual tests

### CODE REVIEW

**Comment by yaooqinn** [comment]
ping @srowen would you plz take a look at this pr？

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala [hunk 1]
```diff
@@ -34,28 +30,13 @@ import org.apache.spark.sql.hive.thriftserver.server.SparkSQLOperationManager
 
 
 private[hive] class SparkSQLSessionManager(hiveServer: HiveServer2, sqlContext: SQLContext)
-  extends SessionManager(hiveServer)
-  with ReflectedCompositeService {
```
Comment (gatorsmile) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala]: What is the reason this PR removes `ReflectedCompositeService`?

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala [hunk 2]
```diff
@@ -34,28 +30,13 @@ import org.apache.spark.sql.hive.thriftserver.server.SparkSQLOperationManager
...
-
-    val backgroundPoolSize = hiveConf.getIntVar(ConfVars.HIVE_SERVER2_ASYNC_EXEC_THREADS)
-    setSuperField(this, "backgroundOperationPool", Executors.newFixedThreadPool(backgroundPoolSize))
-    getAncestorField[Log](this, 3, "LOG").info(
-      s"HiveServer2: Async execution pool size $backgroundPoolSize")
-
     setSuperField(this, "operationManager", sparkSqlOperationManager)
-    addService(sparkSqlOperationManager)
-
-    initCompositeService(hiveConf)
```
Comment (gatorsmile) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala]: Why removing this line?

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala [hunk 3]
```diff
@@ -34,28 +30,13 @@ import org.apache.spark.sql.hive.thriftserver.server.SparkSQLOperationManager
...
-      invoke(classOf[SessionManager], this, "initOperationLogRootDir")
-    }
-
-    val backgroundPoolSize = hiveConf.getIntVar(ConfVars.HIVE_SERVER2_ASYNC_EXEC_THREADS)
-    setSuperField(this, "backgroundOperationPool", Executors.newFixedThreadPool(backgroundPoolSize))
-    getAncestorField[Log](this, 3, "LOG").info(
-      s"HiveServer2: Async execution pool size $backgroundPoolSize")
-
     setSuperField(this, "operationManager", sparkSqlOperationManager)
-    addService(sparkSqlOperationManager)
```
Comment (gatorsmile) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala]: This line is also needed, right?

**Comment by HyukjinKwon** [comment]
Hi @yaooqinn, is this PR active? if so, would you address or answer to the review comments above?

**Pull request closed without merging**